LeaderboardBackend/Controllers/AccountController.cs

```diff
@@ -74,4 +74,60 @@ public async Task<ActionResult<UserViewModel>> Register([FromBody] RegisterReque
                 return Conflict(new ValidationProblemDetails(ModelState));
             });
     }
+
+    /// <summary>
+    ///     Logs a User in.
+    /// </summary>
+    /// <param name="request">
+    ///     The `LoginRequest` instance from which to perform the login.
+    /// </param>
+    /// <response code="200">
+    ///     The `User` was logged in successfully. A `LoginResponse` is returned, containing a token.
+    /// </response>
+    /// <response code="400">The request was malformed.</response>
+    /// <response code="401">The password given was incorrect.</response>
+    /// <response code="403">The associated `User` is banned.</response>
+    /// <response code="404">No `User` with the requested details could be found.</response>
+    /// <response code="422">
+    ///     The request contains errors.<br/><br/>
+    ///     Validation error codes by property:
+    ///     - **Password**:
+    ///       - **NotNullValidator**: No password was passed
+    ///       - **PasswordFormat**: Invalid password format
+    ///     - **Email**:
+    ///       - **NotNullValidator**: No email was passed
+    ///       - **EmailValidator**: Invalid email format
+    /// </response>
+    [AllowAnonymous]
+    [HttpPost("/login")]
+    [ApiConventionMethod(typeof(Conventions), nameof(Conventions.PostAnon))]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+#pragma warning disable CS1573 // Hides warning for not having authService in the XML comment above
+    public async Task<ActionResult<LoginResponse>> Login([FromBody] LoginRequest request, [FromServices] IAuthService authService)
+    {
```

Comment:
business logic should be in a service class

Reply:
I've made a new method `GetUserByEmailAndPassword` as I'd need to change the function signature, and the existing method already has use in the `Me()` action.